### PR TITLE
[Shallow] Fix some errors

### PR DESCRIPTION
**Pull request by miguelmaso**
**📝 Description**
This PR fixes almost all the errors identified in #9597 . Most of the bugs are silently hidden warnings in the pc.
There is still a memory error to fix.

**🆕 Changelog**
- Fix a lot of errors caught by ci

### CODE REVIEW

**Comment by miguelmaso** [comment]
@ipouplana can you review it, please?

**Review by ipouplana — APPROVED** [review]
It seems good to me

File: applications/ShallowWaterApplication/custom_elements/wave_element.h [hunk 1]
```diff
@@ -335,11 +335,11 @@ class WaveElement : public Element
         const std::size_t I,
         const std::size_t J);
 
-    static const array_1d<double,3> VectorProduct(
+    static array_1d<double,3> VectorProduct(
```
Comment (rubenzorrilla) [inline applications/ShallowWaterApplication/custom_elements/wave_element.h:338]: I'd have done this method `const`.

File: applications/ShallowWaterApplication/custom_elements/wave_element.h [hunk 2]
```diff
@@ -335,11 +335,11 @@ class WaveElement : public Element
...
         const array_1d<array_1d<double,3>,TNumNodes>& rV,
         const array_1d<double,TNumNodes>& rN);
 
-    static const double VectorProduct(
+    static double VectorProduct(
```
Comment (rubenzorrilla) [inline applications/ShallowWaterApplication/custom_elements/wave_element.h:342]: Same here.

File: applications/ShallowWaterApplication/custom_elements/wave_element.h (hunk 1 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/custom_elements/wave_element.h:338]: I do not really understand all the stuff related to const. Correct me if I'm wrong:
`static const array_1d VectorProduct(...);` -> The output is declared const. This is ignored by the compiler, the CI throws a warning
`static array_1d VectorProduct(...) const;` -> A static method can't be marked as const (it does not have the `this` pointer). Compilation error.

Comment (rubenzorrilla) [reply applications/ShallowWaterApplication/custom_elements/wave_element.h:338]: About 1, this is ignored as const must be declared outside (when assigning the return value). About 2, sorry, I didn't note these being static. Forget about my comment.

**Pull request merged**